app/controllers/api/braintree_controller.rb

```diff
@@ -4,7 +4,26 @@ def token
   end
 
   def transaction
-    # we could use permitted params instead of params.except(:action, :controller)
-    render json: PaymentProcessor::Clients::Braintree::Transaction.make_transaction(params.except(:action, :controller))
+    sale = braintree.make_transaction(options)
+
+    if sale.success?
+      render json: { success: true, transaction_id: sale.transaction_id }
+    else
+      render json: { success: false, errors: [] }
+    end
+  end
```

Comment:
@Tuuleh I thought it'd be nice if the `sale` object had a `success` flag so we can clearly show our reasoning about what to send back to the client. Also, Braintree returns a unique `transaction_id` (https://developers.braintreepayments.com/reference/response/transaction/ruby#id ) for a successful sale - I think we should pass that on to the client.